pkg/sql/copy/copy_test: TestLargeDynamicRows failed

pkg/sql/copy/copy_test.TestLargeDynamicRows [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11417357?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11417357?buildTab=artifacts#/) on release-23.1 @ [ccd59288fdb10e2f2c1e562b2699484ace41148b](https://github.com/cockroachdb/cockroach/commits/ccd59288fdb10e2f2c1e562b2699484ace41148b):


```
=== RUN   TestLargeDynamicRows
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/53491fdf16955d5896449a680358dad1/logTestLargeDynamicRows385641696
    test_log_scope.go:79: use -show-logs to present logs inline
    copy_test.go:612: 
        	Error Trace:	pkg/sql/copy/copy_test/pkg/sql/copy/copy_test.go:612
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestLargeDynamicRows
    panic.go:522: -- test log scope end --
--- FAIL: TestLargeDynamicRows (1.93s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeDynamicRows.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30796

--- a/pkg/sql/copy/copy_test.go
+++ b/pkg/sql/copy/copy_test.go
@@ -601,9 +601,10 @@ func TestLargeDynamicRows(t *testing.T) {
 	err = conn.Exec(ctx, "RESET CLUSTER SETTING kv.raft.command.max_size")
 	require.NoError(t, err)
 
-	// This won't work if the batch size gets set to less than 4.
-	if sql.CopyBatchRowSize < 4 {
-		sql.SetCopyFromBatchSize(4)
+	// This won't work if the batch size gets set to less than 5. When the batch
+	// size is 4, the test hook will count an extra empty batch.
+	if sql.CopyBatchRowSize < 5 {
+		sql.SetCopyFromBatchSize(5)
 	}
 
 	_, err = conn.GetDriverConn().CopyFrom(ctx, strings.NewReader(sb.String()), "COPY t FROM STDIN")

